dataRetriever/resolvers/receiptResolver.go

```diff
@@ -0,0 +1,187 @@
+			topic:            arg.SenderResolver.RequestTopic(),
+			throttler:        arg.Throttler,
+		},
+	}
+
+	return rcResolver, nil
+}
+
+// ProcessReceivedMessage will be the callback func from the p2p.Messenger and will be called each time a new message was received
+// (for the topic this validator was registered to, usually a request topic)
+func (rcRes *receiptResolver) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPeer core.PeerID, source p2p.MessageHandler) error {
+	err := rcRes.canProcessMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	rcRes.throttler.StartProcessing()
+	defer rcRes.throttler.EndProcessing()
+
+	rd, err := rcRes.parseReceivedMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	switch rd.Type {
+	case dataRetriever.HashType:
+		err = rcRes.resolveReceiptRequestByHash(rd.Value, message.Peer(), rd.Epoch, source)
+	case dataRetriever.HashArrayType:
+		err = rcRes.resolveReceiptRequestByHashArray(rd.Value, message.Peer(), rd.Epoch, source)
+	default:
+		err = dataRetriever.ErrRequestTypeNotImplemented
+	}
+
+	if err != nil {
+		err = fmt.Errorf("%w for hash %s", err, logger.DisplayByteSlice(rd.Value))
+	}
+
+	return err
+}
+
+func (rcRes *receiptResolver) resolveReceiptRequestByHash(hash []byte, pid core.PeerID, epoch uint32, source p2p.MessageHandler) error {
+	receipt, err := rcRes.fetchReceiptAsByteSlice(hash, epoch)
+	if err != nil {
+		return err
+	}
+
+	b := &batch.Batch{
+		Data: [][]byte{receipt},
+	}
+	buffToSend, err := rcRes.marshalizer.Marshal(b)
+	if err != nil {
+		return err
+	}
+
+	return rcRes.Send(buffToSend, pid, source)
+}
+
+func (rcRes *receiptResolver) fetchReceiptAsByteSlice(hash []byte, epoch uint32) ([]byte, error) {
+	value, ok := rcRes.receiptPool.Peek(hash)
+	if ok {
+		return rcRes.marshalizer.Marshal(value)
+	}
+
+	buff, err := rcRes.getFromStorage(hash, epoch)
+	if err != nil {
+		rcRes.DebugHandler().LogFailedToResolveData(
+			rcRes.topic,
+			hash,
+			err,
+		)
+
+		return nil, err
+	}
+
+	rcRes.DebugHandler().LogSucceededToResolveData(rcRes.topic, hash)
+
+	return buff, nil
+}
+
+func (rcRes *receiptResolver) resolveReceiptRequestByHashArray(mbBuff []byte, pid core.PeerID, epoch uint32, source p2p.MessageHandler) error {
+	b := batch.Batch{}
+	err := rcRes.marshalizer.Unmarshal(&b, mbBuff)
+	if err != nil {
+		return err
+	}
+	hashes := b.Data
+
+	var errFetch error
+	errorsFound := 0
+	mbsBuffSlice := make([][]byte, 0, len(hashes))
+	for _, hash := range hashes {
+		mb, errTemp := rcRes.fetchReceiptAsByteSlice(hash, epoch)
+		if errTemp != nil {
+			errFetch = fmt.Errorf("%w for hash %s", errTemp, logger.DisplayByteSlice(hash))
+			log.Trace("fetchMbAsByteSlice missing",
+				"error", errFetch.Error(),
+				"hash", hash)
+			errorsFound++
+
+			continue
+		}
+		mbsBuffSlice = append(mbsBuffSlice, mb)
+	}
+
+	buffsToSend, errPack := rcRes.dataPacker.PackDataInChunks(mbsBuffSlice, maxBuffToSendBulkMiniblocks)
```

Comment:
```suggestion
	buffsToSend, errPack := rcRes.dataPacker.PackDataInChunks(rcpBuffSlice, maxBuffToSendBulkMiniblocks)
```

Reply:
perhaps add a new const instead of maxBuffToSendBulkMiniblocks, specific to receipt?